sdk/schemaregistry/Azure.Data.SchemaRegistry/src/Serialization/SchemaRegistrySerializer.cs

```diff
@@ -0,0 +1,570 @@
+            if (contentTypeArray.Length != 2 || contentTypeArray[0] != _mimeType)
+            {
+                throw new FormatException($"Content type was not in the expected format of '{_mimeType}+schema-id', where 'schema-id' " +
+                                          "is the Schema Registry schema ID.");
+            }
+
+            string schemaId = contentTypeArray[1];
+
+            if (async)
+            {
+                return await DeserializeInternalAsync(data, dataType, schemaId, true, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                return DeserializeInternalAsync(data, dataType, schemaId, false, cancellationToken).EnsureCompleted();
+            }
+        }
+
+        private async ValueTask<object> DeserializeInternalAsync(
+            BinaryData data,
+            Type dataType,
+            string schemaId,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            Argument.AssertNotNull(schemaId, nameof(schemaId));
+
+            string schemaDefinition;
+            try
+            {
+                // Attempt to get the schema definition using the schema Id from the Content Type
+                if (async)
+                {
+                    schemaDefinition = await GetSchemaByIdAsync(schemaId, true, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    schemaDefinition = GetSchemaByIdAsync(schemaId, false, cancellationToken).EnsureCompleted();
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"An error occurred while attempting to retrieve the schema with id {schemaId}.", ex);
+            }
+
+            var deserializer = _serializerOptions.Serializer;
+            object objectToReturn;
+            try
+            {
+                // Attempt to deserialize
+                var dataStream = data.ToStream();
+                if (async)
+                {
+                    objectToReturn = await deserializer.DeserializeAsync(dataStream, dataType, cancellationToken: cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    objectToReturn = deserializer.Deserialize(dataStream, dataType, cancellationToken);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"An error occurred while attempting to deserialize the data.", ex);
+            }
+
+            try
+            {
+                // Attempt to validate
+                _schemaValidator.Validate(objectToReturn, dataType, schemaDefinition);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("The validate method determined the object was invalid according to the schema.", ex);
```

Comment:
Should this message be included in the AggregateException instead - https://github.com/Azure/azure-sdk-for-net/pull/36926/files#diff-52d063411232ba155aa286ba7fc64c1a2e5f6d5fb636d8dbd07743ff91cc062fR42?